web-console/src/views/explore-view/modules/pie-chart-echarts-module.ts

```diff
@@ -20,8 +20,29 @@ import { C, SqlExpression } from '@druid-toolkit/query';
 import { typedVisualModule } from '@druid-toolkit/visuals-core';
 import * as echarts from 'echarts';
 
+import { highlightStore } from '../highlight-store/highlight-store';
 import { getInitQuery } from '../utils';
 
+/**
+ * Returns the cartesian coordinates of a pie slice external centroid
+ */
+function getCentroid(chart: echarts.ECharts, dataIndex: number) {
+  // see these underscores everywhere? that's because those are private properties
+  // I have no real choice but to use them, because there is no public API for this (on pie charts)
+  // #no_ragrets
+  const layout = (chart as any)._chartsViews[0]._data._itemLayouts[dataIndex];
```

Comment:
Maybe for some extra safety in case the private API changes... tradeoff between crashing and not calculating a centroid 🤷 

```suggestion
  const layout = (chart as any)._chartsViews?.[0]?._data?._itemLayouts?.[dataIndex];
```